feat(ingestion/bigquery): BigQuery Owner Label to Datahub Ownership (#10047)
datahub-project/datahub: metadata-ingestion/docs/transformer/dataset_transformer.md

### Config Details
| Field                       | Required | Type    | Default       | Description                                 |
|-----------------------------|----------|---------|---------------|---------------------------------------------|
| `tag_pattern`  |          | str     |               | Regex to use for tags to match against. Supports Regex to match a pattern which is used to remove content. Rest of string is considered owner ID for creating owner URN. |
| `is_user`      |          | bool    | `true`   | Whether should be consider a user or not. If `false` then considered a group. |
| `owner_character_mapping` |          | dict[str, str]  |     | A mapping of extracted owner character to datahub owner character. |
| `email_domain` |          | str    |    | If set then this is appended to create owner URN. |

> | `extract_owner_type_from_tag_pattern` |          | str    |  `false`   | Whether to extract an owner type from provided tag pattern first group. If `true`, no need to provide owner_type and owner_type_urn config. For example: if provided tag pattern is `(.*)_owner_email:` and actual tag is `developer_owner_email`, then extracted owner type will be `developer`.|

Reviewer: please update the examples in the docs to demonstrate how these configs should be used

| `email_domain` |          | str    |    | If set then this is appended to create owner URN. |
| `extract_owner_type_from_tag_pattern` |          | str    |  `false`   | Whether to extract an owner type from provided tag pattern first group. If `true`, no need to provide owner_type and owner_type_urn config. For example: if provided tag pattern is `(.*)_owner_email:` and actual tag is `developer_owner_email`, then extracted owner type will be `developer`.|
| `owner_type` |          | str    |  `TECHNICAL_OWNER`   | Ownership type. |
| `owner_type_urn` |          | str    |  `None`   | Set to a custom ownership type's URN if using custom ownership. |

Let’s suppose we’d like to add a dataset ownerships based on part of dataset tags. To do so, we can use the `extract_ownership_from_tags` transformer that’s included in the ingestion framework.

The config, which we’d append to our ingestion recipe YAML, would look like this:

```yaml
transformers:
  - type: "extract_ownership_from_tags"
    config:
      tag_pattern: "owner_email:"
```

So if we have input dataset tag like
- `urn:li:tag:dataset_owner_email:abc@email.com`

> - `urn:li:tag:dataset_owner_email:xyz@email.com`

Reviewer: these have `dataset_owner_email` but the example is `tag_pattern: "owner_email:"` - why the mismatch?

Author: Because now the config is tag_pattern and not tag_prefix. We can provide pattern in this way as well.

> String after the matched tag pattern will be considered as owner to create ownership. Hence an owners called `abc@email.com` and `xyz@email.com` will be added to them respectively.

Reviewer: ```suggestion
The portion of the tag after the matched tag pattern will be converted into an owner. Hence users `abc@email.com` and `xyz@email.com` will be added as owners.
```

| Field                       | Required | Type    | Default       | Description                                 |
|-----------------------------|----------|---------|---------------|---------------------------------------------|

> | `semantics`                 |          | enum    | `OVERWRITE`   | Whether to OVERWRITE or PATCH the entity present on DataHub GMS. |

Reviewer: is this field respected? if yes, what does it do?

imo it seems like we should not inherit from `TransformerSemanticsConfigModel`

| `tag_pattern`  |          | str     |               | Regex to use for tags to match against. Supports Regex to match a pattern which is used to remove content. Rest of string is considered owner ID for creating owner URN. |
| `is_user`      |          | bool    | `true`   | Whether should be consider a user or not. If `false` then considered a group. |
| `owner_character_mapping` |          | dict[str, str]  |     | A mapping of extracted owner character to datahub owner character. |
| `email_domain` |          | str    |    | If set then this is appended to create owner URN. |

> | `extract_owner_type_from_tag_pattern` |          | str    |  `false`   | Whether to extract an owner type from provided tag pattern first group. If `true`, no need to provide owner_type and owner_type_urn config.|

Reviewer: provide examples for how this works - it's not super clear
